models/issues/issue.go

```diff
@@ -501,6 +502,60 @@ func GetIssueByIndex(ctx context.Context, repoID, index int64) (*Issue, error) {
 	return issue, nil
 }
 
+func isPullToCond(isPull optional.Option[bool]) builder.Cond {
+	if isPull.Has() {
+		return builder.Eq{"is_pull": isPull.Value()}
+	}
+	return builder.NewCond()
+}
+
+func FindLatestIssues(ctx context.Context, repoID int64, isPull optional.Option[bool], pageSize int) (IssueList, error) {
+	issues := make([]*Issue, 0, pageSize)
+	err := db.GetEngine(ctx).Where("repo_id = ?", repoID).
+		And(isPullToCond(isPull)).
+		OrderBy("created_unix DESC").
+		Limit(pageSize).
+		Find(&issues)
+	return issues, err
+}
+
+func FindIssuesTitleKeywords(ctx context.Context, repoID int64, keyword string, isPull optional.Option[bool], nonIDs []int64, pageSize int) (IssueList, error) {
+	issues := make([]*Issue, 0, pageSize)
+	sess := db.GetEngine(ctx).Where("repo_id = ?", repoID).
+		And(isPullToCond(isPull))
+	if len(nonIDs) > 0 {
+		sess.NotIn("id", nonIDs)
+	}
+	err := sess.And("name LIKE ?", "%"+keyword+"%").
+		OrderBy("created_unix DESC").
+		Limit(pageSize).
+		Find(&issues)
+	return issues, err
+}
+
+func FindIssuesWithIndexPrefix(ctx context.Context, repoID, index int64, isPull optional.Option[bool], pageSize int) (IssueList, error) {
+	var cond string
+	switch {
+	case setting.Database.Type.IsSQLite3():
+		cond = "CAST(`index` AS TEXT) LIKE ?"
+	case setting.Database.Type.IsMySQL():
+		cond = "CAST(`index` AS CHAR) LIKE ?"
+	case setting.Database.Type.IsPostgreSQL():
+		cond = "index::TEXT LIKE ?"
+	case setting.Database.Type.IsMSSQL():
+		cond = "CAST([index] AS VARCHAR) LIKE ?"
+	}
```

Comment:
I do not think we should do so.

Just query the issue full id, do not query the prefix (why a user would expect to use `#2` to list `#2354`? They only expects to see `#2` issue)

Reply:
Because i.e. if he wants to input `#360`, he needs to input the number one by one. When he input `#3`, or `#36`, we can list all the possible results `#3x` or `#3xx`. I can remove this if you don't think it's useful.

Reply:
Are you sure that the list of `#3` or `#36` really helps? There are more than hundreds of candidate results for `#3`, does it help? Who would really like to pick an issue from that candidate list?

Reply:
It should just list the issue matching the number on the top, that's enough.

If user inputs `#3`, show issue `#3` as the first item of the suggestion list.

Reply:
Suppose the issue you'd like to pick is `#33538` (this one's number).

You mean that you can't remember `33538` but can only remember `3353`? If you only input `#3` or `#33`, are you able to see the `#33538` in the list?

Reply:
f37f3897d04a8eaf51eabdb45818fe4a3b88e84e